eng/scripts/ValidateProjectCoverage.ps1

```diff
@@ -0,0 +1,199 @@
+﻿#!/usr/bin/env pwsh
+<#
+.SYNOPSIS
+    Validates the code coverage policy for each project.
+.DESCRIPTION
+    This script compares code coverage with thresholds given in "MinCodeCoverage" property in each project.
+    The script writes an error for each project that does not comply with the policy.
+.PARAMETER CoberturaReportXml
+    Path to the XML file to read the code coverage report from in Cobertura format
+.EXAMPLE
+    PS> .\ValidatePerProjectCoverage.ps1 -CoberturaReportXml .\Cobertura.xml
+#>
+
+param (
+    [Parameter(Mandatory = $true, HelpMessage="Path to the XML file to read the code coverage report from")]
+    [string]$CoberturaReportXml
+)
+
+function Write-Header {
+    param($message, [bool]$isError);
+    $color = if ($isError) { 'Red' } else { 'Green' };
+    Write-Host $message -ForegroundColor $color;
+    Write-Host ("=" * 80)
+ }
+function Get-XmlValue { param($X, $Y); return $X.SelectSingleNode($Y).'#text' }
+
+Write-Verbose "Reading cobertura report..."
+[xml]$CoberturaReport = Get-Content $CoberturaReportXml
+if ($null -eq $CoberturaReport.coverage -or 
+    $null -eq $CoberturaReport.coverage.packages -or
+    $null -eq $CoberturaReport.coverage.packages.package -or 
+    0 -eq $CoberturaReport.coverage.packages.package.count)
+{
+    return
```

Comment:
When can this happen? Should this also generate a comment if it means that code coverage did not get generated for some reason?

Reply:
Hard to say. Some of this infra originated from R9, I can speculated that there may have been some weird situations where something like that happened... 
How likely this is to happen, probably unlikely but again, we don't control the subsystem that generates cobertura files ¯\\_(ツ)\_/¯

We can certainly emit a comment at this point, and fail a build (when we'll be failing a build). Good idea 👍

Reply:
How does [this look](https://github.com/dotnet/aspire/pull/4313#issuecomment-2172308578)?
![image](https://github.com/dotnet/aspire/assets/4403806/096430ec-7f4f-428f-9c81-8bdbbd8b3e35)

Reply:
This would catch the case where the code coverage report generation did not *fail* but emitted an empty file.

Reply:
We don't control coverage reporting, it's essentailly a 3P tool as far as this CI is concerned, so anything can happen. If the coverage generation fails it would likely fail the build at the previous step. If the coverage file was produced and we got to this point, then we'd fail here, if the content was unexpected.
I think that's good enough for now, we can always tweak it later, if desired.